src/services/document-service.js

```diff
@@ -69,6 +71,26 @@ const getTagsQuery = searchString => [
 
 const lastUpdatedFirst = [['updatedOn', -1]];
 
+const createNewDocumentRevision = ({ doc, documentKey, userId, nextOrder, restoredFrom, sections }) => {
+  logger.info('Creating new revision for document key %s with order %d', documentKey, nextOrder);
+
+  return {
+    _id: uniqueId.create(),
+    key: documentKey,
+    order: nextOrder,
+    restoredFrom,
+    createdOn: dateTime.now(),
+    createdBy: userId,
+    title: doc.title || '',
+    slug: doc.slug || '',
+    namespace: doc.namespace,
+    language: doc.language,
+    sections: sections?.length ? sections : doc.sections,
```

Comment:
What if I want the empty array?

Reply:
That's true, you could have deleted all sections in one revision. Corrected, thanks!